### PR TITLE
Added ShouldThrow and ShouldNotThrow extensions

### PR DESCRIPTION
**Pull request by JoeMighty**
Here are the ShouldThrow and ShouldNotThrow extensions from for issue #213. I've updated the tests accordingly - I hope they're all ok!

### CODE REVIEW

File: src/Shouldly/ShouldStaticClasses/ShouldThrowExtensions.cs [hunk 1]
```diff
@@ -8,6 +8,20 @@ namespace Shouldly
     [ShouldlyMethods]
     public static partial class Should
     {
+        /*** ShouldThrow(Action) ***/
+        public static TException ShouldThrow<TException>(this Action actual) where TException : Exception
```
Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensions.cs]: I think we should put these extensions in another class so we don't have `Should.ShouldThrow` and `Should.Throw` on the same class.

Maybe move it `ShouldThrowExensions` or something

**Comment by JakeGinnivan** [comment]
Just the one comment. Looks awesome apart from that

**Comment by JoeMighty** [comment]
Great! I'm just about to head off to bed now but I'll try move them into another class tomorrow during my lunch break.

**Comment by JakeGinnivan** [comment]
:+1:

**Comment by JoeMighty** [comment]
As `ShouldThrowExtensions.cs` already exists I moved the newly created overloads into classes respective Overload.cs classes (ie: `ShouldThrowExtensionsOverloads.cs`). I hope this is ok, if not then let me know and I'll update it.

For some reason it appears to have broken the AppVeyor build but I can't see why it would have done this. It's complaining about references to tasks?

**Comment by JakeGinnivan** [comment]
Yeah, Shouldly is both .net 3.5 and .net 4.0. So you need the `#if net40` directtives around any code which references .net 4.0 stuff.

**Comment by JakeGinnivan** [comment]
The solution opens as .net 4.0 only. you can run `msbuild shouldly.sln /p:targetframeworkversion=v3.5` to test locally

File: src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using JetBrains.Annotations;
+
+namespace Shouldly
+{
+    public static partial class Should
```
Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: Change the class name as well, these methods still exist on the `Should` type.

Comment (JoeMighty) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: Good point, is this better:

    public static class ShouldThrowOverloads
    {
        public static TException ShouldThrow<TException>(this Action actual) where TException : Exception
        {
            return Should.Throw<TException>(actual);
        }
    ...

Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: yep, that would be great

Comment (JoeMighty) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: It looks like updating the extension to return `Should.Throw` will result in the error message changing from "action should throw" to "return Should throw". It'd be nice to keep the variable (`action`) in the message. Is it worth looking at a custom message generator or do you point me in the direction of another way around this problem that allows us to move these methods into their own class yet retain the helpful error message?

Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: Make sure you attribute the new class with 
```
[DebuggerStepThrough]
[ShouldlyMethods]
```

Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: Im heading to bed (back in Australia now). If you don't make progress let me know, I will merge then have a look at solving in the morning when I get up

Comment (JakeGinnivan) [inline src/Shouldly/ShouldStaticClasses/ShouldThrowExtensionsOverloads.cs]: The PR looks great, i'm sure I can sort the small remaining issues

**Comment by JakeGinnivan** [comment]
Have time this morning to do some OSS work, figured i'd get this merged as you have done all the hard work :) Thanks!

**Comment by JoeMighty** [comment]
Alright, thanks for picking that up for me.

**Comment by JakeGinnivan** [comment]
Not a prob, was hardly anything to do didn't seem worth another round trip :+1:

**Pull request merged**